joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/astcreation/AstCreator.scala

```diff
@@ -1903,16 +1906,13 @@ class AstCreator(filename: String, global: Global)
           callNode.name == "require_once" ||
           callNode.name == "load"
         ) {
-          val importedFile =
-            argsAsts.head.nodes
-              .filter(node => node.isInstanceOf[NewLiteral])
-              .head
-              .asInstanceOf[NewLiteral]
-              .code
-          println(s"AST to be created for imported file ${importedFile}")
-
+          val importedNode =
+            argsAsts.head.nodes.filter(node => node.isInstanceOf[NewLiteral]).head.asInstanceOf[NewLiteral]
+          println(s"AST to be created for imported file ${importedNode.code}")
+          Seq(callAst(callNode.code(importedNode.code), Seq(Ast(importedNode))))
```

Comment:
Why do we create CallAst for importedNodes? we should create ImportNodes right?

Reply:
Not sure about this but i think to parse the complicated require statement like: `require File.join(__dir__, 'file.rb')`

@rahul-privado can you help?

Reply:
Makes sense. Did not know about them.

@ankit-privado Can you change this and use `newImportNode()`

Reply:
yes sure @rahul-privado

Reply:
done.